packages/website/src/components/DocLayout.js

```diff
@@ -121,3 +57,87 @@ function DocLayout({ children, pathname }) {
 }
 
 export default DocLayout;
+
+const Sidebar = ({ data, pathname }) => {
+  const sidebarRef = useRef(null);
+  const { current, setPrevious } = useContext(SidebarScrollContext);
+
+  useEffect(() => {
```

Comment:
using `useLayoutEffect` here would be better as it makes sure that the user will not see the scroll before it gets updated to a new scrolling position.